Providing more information while crashing process in async error handling

osalpekar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#46274 Providing more information while crashing process in async error handling**

We crash the process in NCCL Async Error Handling if the collective
has been running for greater than some set timeout. This PR logs more
information about the rank and duration the collective ran before throwing an exception.

Differential Revision: [D24200144](https://our.internmc.facebook.com/intern/diff/D24200144/)

osalpekar committed:
Providing more information while crashing process in async error handling

We crash the process in NCCL Async Error Handling if the collective
has been running for greater than some set timeout. This PR introduces more
information about the rank and duration the collective ran - we will also
include details about which collective timed out once #46010 lands.

Differential Revision: [D24200144](https://our.internmc.facebook.com/intern/diff/D24200144/)

[ghstack-poisoned]

osalpekar committed:
Update on "Providing more information while crashing process in async error handling"


We crash the process in NCCL Async Error Handling if the collective
has been running for greater than some set timeout. This PR introduces more
information about the rank and duration the collective ran - we will also
include details about which collective timed out once #46010 lands.

Differential Revision: [D24200144](https://our.internmc.facebook.com/intern/diff/D24200144/)

[ghstack-poisoned]

osalpekar committed:
Update on "Providing more information while crashing process in async error handling"


We crash the process in NCCL Async Error Handling if the collective
has been running for greater than some set timeout. This PR introduces more
information about the rank and duration the collective ran - we will also
include details about which collective timed out once #46010 lands.

Differential Revision: [D24200144](https://our.internmc.facebook.com/intern/diff/D24200144/)

[ghstack-poisoned]

pritamdamania87 commented on torch/lib/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -608,12 +614,17 @@ void ProcessGroupNCCL::ncclCommWatchdogInternal() {
         // Check for Timeouts in the WorkNCCL Operations, and abort all
         // communicators accordingly.
         if (work.timedOut()) {
-          LOG(INFO)
-              << "[Rank " << rank_
-              << "] Watchdog caught collective operation timeout for work: "
-              << work;
```
Shouldn't we retrain the LOG message here?

osalpekar replied:
Added back the log in the workCleanupThread

osalpekar committed:
Update on "Providing more information while crashing process in async error handling"


We crash the process in NCCL Async Error Handling if the collective
has been running for greater than some set timeout. This PR logs more
information about the rank and duration the collective ran before throwing an exception.

Differential Revision: [D24200144](https://our.internmc.facebook.com/intern/diff/D24200144/)

[ghstack-poisoned]

osalpekar committed:
Update on "Providing more information while crashing process in async error handling"


We crash the process in NCCL Async Error Handling if the collective
has been running for greater than some set timeout. This PR logs more
information about the rank and duration the collective ran before throwing an exception.

Differential Revision: [D24200144](https://our.internmc.facebook.com/intern/diff/D24200144/)

[ghstack-poisoned]

osalpekar committed:
Update on "Providing more information while crashing process in async error handling"


We crash the process in NCCL Async Error Handling if the collective
has been running for greater than some set timeout. This PR logs more
information about the rank and duration the collective ran before throwing an exception.

Differential Revision: [D24200144](https://our.internmc.facebook.com/intern/diff/D24200144/)

[ghstack-poisoned]
